frontend/src/component/project/Project/ProjectSettings/ProjectActions/ProjectActionsTable/ProjectActionsModal/ProjectActionsForm/ActionItem.tsx

```diff
@@ -0,0 +1,134 @@
+import { IconButton, Tooltip, styled } from '@mui/material';
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+import { IAction } from 'interfaces/action';
+import { Fragment } from 'react';
+import GeneralSelect, {} from 'component/common/GeneralSelect/GeneralSelect';
+import { Delete } from '@mui/icons-material';
+import { useProjectEnvironments } from 'hooks/api/getters/useProjectEnvironments/useProjectEnvironments';
+import { useRequiredPathParam } from 'hooks/useRequiredPathParam';
+import mapValues from 'lodash.mapvalues';
+import { useFeatureSearch } from 'hooks/api/getters/useFeatureSearch/useFeatureSearch';
+import {
+    BoxSeparator,
+    Col,
+    InnerBoxHeader,
+    Row,
+    StyledInnerBox,
+} from './InnerContainerBox';
+
+export type UIAction = Omit<IAction, 'id' | 'createdAt' | 'createdByUserId'> & {
+    id: string;
+};
+
+export const Action = ({
+    action,
+    index,
+    stateChanged,
+    onDelete,
+}: {
+    action: UIAction;
+    index: number;
+    stateChanged: (action: UIAction) => void;
+    onDelete: () => void;
+}) => {
+    const { id, action: actionName } = action;
+    const projectId = useRequiredPathParam('projectId');
+    const environments = useProjectEnvironments(projectId);
+    const { features } = useFeatureSearch(
+        mapValues(
+            {
+                project: `IS:${projectId}`,
+            },
+            (value) => (value ? `${value}` : undefined),
+        ),
+        {},
+    );
+    return (
+        <Fragment key={id}>
+            <ConditionallyRender
+                condition={index > 0}
+                show={<BoxSeparator text='THEN' />}
+            />
+            <StyledInnerBox>
+                <Row>
+                    <span>Action {index + 1}</span>
+                    <InnerBoxHeader>
+                        <Tooltip title='Delete action' arrow>
+                            <IconButton type='button' onClick={onDelete}>
+                                <Delete />
+                            </IconButton>
+                        </Tooltip>
+                    </InnerBoxHeader>
+                </Row>
+                <Row>
+                    <Col>
+                        <GeneralSelect
+                            label='Action'
+                            name='action'
+                            options={[
+                                {
+                                    label: 'Enable flag',
+                                    key: 'TOGGLE_FEATURE_ON',
+                                },
+                                {
+                                    label: 'Disable flag',
+                                    key: 'TOGGLE_FEATURE_OFF',
+                                },
+                            ]}
```

Comment:
We'll probably want to extract this to a constant somewhere.

Reply:
I was actually thinking that the backend should provide the available actions, each one with its required params (which should be optional, because you can either provide them here or dynamically when calling the webhook)